starclass/training_sets/training_set.py

```diff
@@ -47,15 +50,16 @@ def __init__(self, datalevel='corr', tf=0.0):
 		self.train_idx = np.arange(self.nobjects, dtype=int) # Define here because it is needed by self.labels() used below
 		self.test_idx = np.array([], dtype=int)
 		if self.testfraction > 0:
+			labels = np.array([i[0].value for i in self.labels()])
 			self.train_idx, self.test_idx = train_test_split(
 				np.arange(self.nobjects),
 				test_size=self.testfraction,
-				random_state=42,
-				stratify=self.labels()
+				random_state=self.random_seed,
+				stratify=labels, #self.labels()
 			)
-			# Have to sort as train_test_split shuffles and we don't want that
-			self.train_idx = np.sort(self.train_idx)
-			self.test_idx = np.sort(self.test_idx)
+		# Have to sort as train_test_split shuffles and we don't want that
```

Comment:
No reason to change these three lines... It will be sorted if coming directly from `arange`

Reply:
That's true, I put it outside just to make doubly sure that everything is sorted.  `train_test_split` shuffles the data so it needs to be resorted otherwise the wrong labels are picked out later on, so it definitely needs to be sorted inside if `test_fraction != 0`. I will change this now.

Reply:
This should now have been sorted